### ISSUE TITLE
Find/first issue with 3.5

### ISSUE THREAD

**Issue by krtooch** [parse-community/Parse-SDK-JS#1614]
### New Issue Checklist

- [ X] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [ X] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [ X] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).
- [ X] I can reproduce the issue with the latest versions of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse JS SDK](https://github.com/parse-community/Parse-SDK-JS/releases). 

### Issue Description
Unable to user Query.first on 3.5

### Steps to reproduce
Do a first() query with 3.5 client (the query work well with older versions, at least 3.4.2)

### Actual Outcome
giving a collection  'Mission' the query above return an error 

```
const t = await (new Parse.Query('Mission')) .select(['position', 'client']).equalTo('objectId', 'Etk9VWubwP')
           .first()
        console.log(t);
```

outputed error
<img width="681" alt="Capture d’écran 2022-11-18 à 16 57 50" src="https://user-images.githubusercontent.com/18483212/202747314-70f452f7-3f54-410d-83f2-9fc01edf32a5.png">

### Expected Outcome

On lower version
the same query work.
<img width="607" alt="Capture d’écran 2022-11-18 à 16 58 50" src="https://user-images.githubusercontent.com/18483212/202747491-0067956c-cdb2-489b-86c1-dce6c72bfe8b.png">

###  Environment
Issue found on 3.5, work on 3.4.2.

Server
- Parse Server version: `5.3.3`
- Operating system: `Alpine16`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `AWS`

Database
- System (MongoDB or Postgres): `mongo`
- Database version: `4.4`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `ATLAS`

Client
- Parse JS SDK version: `3.5`

**Comment by dblythy** [comment]
3.5.0 is broken, please see #1600. 

Can you try [3.5.1-alpha.2](https://github.com/parse-community/Parse-SDK-JS/releases/tag/3.5.1-alpha.2) or revert to 3.4.4

**Comment by mtrezza** [comment]
I'm closing, if the issue still exists on the latest alpha version please let us know and we'll re-open this issue.

**Comment by dplewis** [comment]
@mtrezza i think we should keep this open until the next stable release. More will come in. The checklist show a check mark for “existing” issues not closed ones

**Comment by mtrezza** [comment]
Yes, let me pin a new issue to the top of the issue list. Our process is that an issue is closed when the PR is merged. This also happens automatically because of the "Closes" tag on GitHub.

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [3.5.1-beta.2](https://github.com/parse-community/Parse-SDK-JS/releases/tag/3.5.1-beta.2)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [3.5.1](https://github.com/parse-community/Parse-SDK-JS/releases/tag/3.5.1)
